frontend/src/components/GoalForm/GoalNudge.js

```diff
@@ -0,0 +1,163 @@
+import {
+  Checkbox,
+} from '@trussworks/react-uswds';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { similiarGoalsByText } from '../../fetchers/goals';
+import { getGoalTemplates } from '../../fetchers/goalTemplates';
+import useAsyncDebounceEffect from '../../hooks/useAsyncDebounceEffect';
+import GoalNudgeText from './GoalNudgeText';
+import GoalNudgeInitiativePicker from './GoalNudgeInitiativePicker';
+
+const MINIMUM_GOAL_NAME_LENGTH = 15;
+
+export const filterOutGrantUsedGoalTemplates = (goalTemplates, selectedGrants) => goalTemplates
+  .filter((template) => {
+    if (!template.goals || !template.goals.length) {
+      return true;
+    }
+    const usedGrantIds = template.goals.map((goal) => goal.grantId);
+    return !selectedGrants.some((grant) => usedGrantIds.includes(grant.id));
+  });
+export default function GoalNudge({
+  error,
+  goalName,
+  validateGoalName,
+  onUpdateText,
+  inputName,
+  isLoading,
+  selectedGrants,
+  recipientId,
+  regionId,
+  onSelectNudgedGoal,
+}) {
+  const [similar, setSimilarGoals] = useState([]);
+  const [useOhsInitiativeGoal, setUseOhsInitiativeGoal] = useState(false);
+  const [dismissSimilar, setDismissSimilar] = useState(false);
+  const [goalTemplates, setGoalTemplates] = useState(null);
+
+  useEffect(() => {
+    if (dismissSimilar) {
+      setSimilarGoals([]);
+    }
+  }, [dismissSimilar]);
+
+  // using DeepCompareEffect to avoid unnecessary fetches
+  // as we have an object (selectedGrants) in the dependency array
+  useDeepCompareEffect(() => {
+    async function fetchGoalTemplates() {
+      try {
+        const templates = await getGoalTemplates(selectedGrants.map((grant) => grant.id));
+        setGoalTemplates(filterOutGrantUsedGoalTemplates(templates, selectedGrants));
+      } catch (err) {
+        setGoalTemplates([]);
+      }
+    }
+
+    if (selectedGrants && selectedGrants.length > 0) {
+      fetchGoalTemplates();
+    }
+  }, [selectedGrants]);
+
+  useAsyncDebounceEffect(async () => {
+    // we need all of these to populate the query
+    if (!recipientId || !regionId || !selectedGrants.length) {
+      return;
+    }
+
+    if (dismissSimilar) {
+      return;
+    }
+
+    try {
+    // we shouldn't run any such query until the user
+      // has typed a minimum number of characters
+      if (goalName.length > MINIMUM_GOAL_NAME_LENGTH) {
+        const similarities = await similiarGoalsByText(
+          regionId,
+          recipientId,
+          goalName,
+          selectedGrants.map((grant) => grant.number),
+        );
+        setSimilarGoals(similarities);
+      }
+    } catch (err) {
+      setSimilarGoals([]);
+    }
+  }, [
+    goalName,
+    regionId,
+    recipientId,
+    selectedGrants,
+    dismissSimilar,
+  ]);
+
+  const onChange = (e) => {
+    onUpdateText(e.target.value);
+  };
+
+  // what a hack
+  const checkboxZed = similar.length && !useOhsInitiativeGoal && !dismissSimilar ? 'z-bottom' : '';
```

Comment:
Stacking context for pseudo-dropdown that houses the goal nudge choices is a little funky here, some of the checkbox was poking through. This just drops it below the form if the dropdown is open